src/routing/stage.rs

```diff
@@ -173,14 +174,28 @@ impl Stage {
         recipients: &[SocketAddr],
         delivery_group_size: usize,
         message: Bytes,
+        to_client: bool,
     ) -> Vec<Command> {
-        self.comm
-            .send_message_to_targets(recipients, delivery_group_size, message)
-            .await
-            .1
-            .into_iter()
-            .map(Command::HandlePeerLost)
-            .collect()
+        if to_client {
+            if self
+                .comm
+                .send_message_to_client(&recipients[0], message)
+                .await
+                .is_err()
+            {
+                vec![Command::HandlePeerLost(recipients[0])]
```

Comment:
Do we need to handle peer lost if it is a client?

Reply:
As I understand it will be handled by above layer?
here we are returning the error.